sos/report/plugins/ssacli.py

```diff
@@ -0,0 +1,50 @@
+# ssacli.py
+
```

Comment:
Plugins need to have the standard banner message in them at the top, optionally with a copyright line above it. Any other plugin will serve as a fine reference.

Reply:
> Couple thoughts below, however the big one here is that this seems to be largely the same as the `hpssm` plugin. The main differences being that `hpssm` just calls `ssacli` directly, rather than using the `/opt/...` path and the commands are all for `slot=0`, rather than iterating over slots.

- My use of `/opt/...` comes from forgetting that HPE actually added a `/usr/bin/ssacli` to the RPM package at some point. So using the full `/opt/` path should no longer be required, it's just a habit I've picked up from when `ssacli` wasn't part of the path.
- `hpssm.py` lists 3 packages `packages = ('ilorest', 'ssacli', 'ssaducli',)` 
  - Only the ssacli plugin contains the binary. 
  - The **ilorest** and **ssaducli** RPM packages do not contain the `ssacli` binary and do not have dependencies requiring `ssacli` RPM be installed with them. 

I believe this would run `ssacli` even when it is not present, generating errors in `sos_log` for the plugin unnecessarily.

> 
> What's the difference between this and `hpssm`? My first thought is the current `hpssm` plugin should simply be extended to handle multiple slots, but I'm also not super familiar with hpssm vs. ssacli from a product/vendor perspective.

Agreed on using existing plugins vs. adding duplicated functionality/intent. I had forgotten (and overlooked) the **hpssm** plugin when reviewing for HPE controllers the other week. I'd be happy to rework this PR into a modification of the existing plugin if nobody minds the change. I also suspect it will be helpful to many, as I have observed systems where slot 0 has no card at all, while slot 1 does and various combinations when more than 1 card is present.


**TL;DR**
 
I am mostly familiar with the **HP SSM** abbreviation being **System Software Manager** which was replaced in 2000 with **Image Assistant**. The plugin says its **Smart Storage Management**, but I cannot find that term used by HPE when referring to Smart Array controllers. The abbreviation I would have expected when searching would have been **HPE SSA** (Smart Storage Administrator). 

Maybe this is a combination of two different abbreviations, or an acronym HPE got rid of enough years ago that even google does not provide a historical reference.

Reply:
Sorry for the delay.

> I believe this would run ssacli even when it is not present, generating errors in sos_log for the plugin unnecessarily.

Generally not a concern for this project to attempt to run non-present commands - it is handled gracefully within `Plugin()`.

> Maybe this is a combination of two different abbreviations, or an acronym HPE got rid of enough years ago that even google does not provide a historical reference.

Interesting. Well, I also cannot find a reference to SSM in the sense of storage management so perhaps we should take the opportunity to realign what this plugin does.

Are `ilorest` and/or `ssaducli` in anyway _related_ to `ssacli`? If not, perhaps we should split `hpssm` into multiple plugins instead. If it helps, we could keep the `hp` prefixing for these plugins (e.g. `hpssa`, `hpilo`, etc...). 

> I'd be happy to rework this PR into a modification of the existing plugin if nobody minds the change.

Let's do that for the `ssacli` commands at least. Depending upon the answer to the above question, feel free to either split out the others into new plugins as well or just stick to the multiple-slots update (and rename as you see fit).

Reply:
> Interesting. Well, I also cannot find a reference to SSM in the sense of storage management so perhaps we should take the opportunity to realign what this plugin does.
> 
> Are ilorest and/or ssaducli in anyway related to ssacli? If not, perhaps we should split hpssm into multiple plugins instead. If it helps, we could keep the hp prefixing for these plugins (e.g. hpssa, hpilo, etc...).

Slightly.

- `ssaducli` is a debug tool for Smart Storage Arrays which comes in a separate RPM. I've rarely installed it and even more rarely used it. 
- `ilorest` is a tool to allow CLI or scripted iLo actions. iLo can reconfigure storage controllers as one area of management.

Neither of these tools are required to investigate physical or logical devices with `ssacli` commands used in the current plugin.

> Let's do that for the ssacli commands at least. Depending upon the answer to the above question, feel free to either split out the others into new plugins as well or just stick to the multiple-slots update (and rename as you see fit).

Sounds good. I'll get something put together for submission this week.